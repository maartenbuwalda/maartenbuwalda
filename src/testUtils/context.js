const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
