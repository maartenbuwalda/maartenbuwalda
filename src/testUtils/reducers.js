export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
