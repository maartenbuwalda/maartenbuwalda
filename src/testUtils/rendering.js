import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}
