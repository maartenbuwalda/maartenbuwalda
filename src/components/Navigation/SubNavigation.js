import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import { ColorContext } from '../../context/color'

const Wrapper = styled.div`
  padding: 0 1.5rem;
  overflow: hidden;
  opacity: 0;
  display: flex;
  flex-direction: column;
  transition: background-color 1s, opacity .6s, max-height .6s, padding .6s;
  max-height: 0;
  background-color: rgba(${({ theme }) => theme.rgb.colorMutedLight}, 0.2);

  a {
    text-decoration: none;
    margin: 0;
    padding: .5rem 1rem .5rem;
    font-size: 1.2rem;
  }
`

const HiddenCheckbox = styled.input`
  display: none;

  &:checked + ${Wrapper} {
    opacity: 1;
    max-height: 20rem;
    padding: 1rem 1.5rem;
  }
`

const SubNavigation = ({ isToggled, links, closeMenu }) => {
  return (
    <ColorContext.Consumer>
      {({ theme }) => (
        <>
          <HiddenCheckbox readOnly checked={isToggled} type="checkbox" id="sub-menu-toggle"/>
          <Wrapper theme={theme}>
            {links.map((x, i) => {
              return (
                <NavLink
                  exact
                  key={i}
                  to={x.pathname}
                  activeClassName="active"
                  onClick={closeMenu}
                >
                  {x.label}
                </NavLink>
              )
            })}
          </Wrapper>
        </>
      )}
    </ColorContext.Consumer>
  )
}

SubNavigation.propTypes = {
  isToggled: PropTypes.bool,
  links: PropTypes.array,
  closeMenu: PropTypes.func,
}

export default SubNavigation
