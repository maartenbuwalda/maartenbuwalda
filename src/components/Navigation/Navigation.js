import React from 'react'
import PropTypes from 'prop-types'
import Card from '../../components/Card'
import SubNavigation from './SubNavigation'
import styled from 'styled-components'
import { routes } from '../../constants/routes'
import { NavLink } from 'react-router-dom'
import colors from '../../constants/colors'
import { sizes } from '../../constants/mediaQueries'
import Icon from '@material-ui/core/Icon'

const Navigation = ({ position }) => {
  const checkbox = React.createRef()
  const currentPath = window.location.hash.replace('#', '')
  const closeMenu = () => {
    checkbox.current.checked = false
  }
  return (
    <Card position={position}>
      <MobileMenuToggle htmlFor="menu-toggle">
        <Icon>menu</Icon>
      </MobileMenuToggle>
      <Menu>
        <HiddenCheckbox
          type="checkbox"
          id="menu-toggle"
          ref={checkbox}
        />
        <LinkWrapper>
          <MobileMenuToggle htmlFor="menu-toggle">
            <Icon>close</Icon>
          </MobileMenuToggle>
          {routes.map((route, i) => {
            const { pathname, subNavigation, icon, label } = route
            return (
              <div key={i}>
                <NavLink
                  exact={pathname === '/'}
                  to={pathname}
                  className="main-link"
                  activeClassName="active"
                  onClick={() => {
                    if (!subNavigation.length) closeMenu()
                  }}
                >
                  <IconWrapper>
                    <Icon>{icon}</Icon>
                  </IconWrapper>
                  <span>{label}</span>
                </NavLink>
                {subNavigation.length > 0 && (
                  <SubNavigation
                    isToggled={currentPath.includes(pathname)}
                    closeMenu={closeMenu}
                    links={subNavigation}
                  />
                )}
              </div>
            )
          })}
        </LinkWrapper>
      </Menu>
    </Card>
  )
}

Navigation.propTypes = {
  position: PropTypes.string,
}

const IconWrapper = styled.div`
  transition: all .3s;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 1rem;
  width: 1rem;
  padding: 1rem;
  background-color: ${colors.gray};
  transform: translateX(-4rem) scaleX(0);
`

const Menu = styled.div`
  border-right: 1px solid ${colors.gray};
  background: ${colors.white};

  @media (min-width: ${sizes.m}) {
    padding: 0;
    position: initial;
    justify-content: flex-start;
    align-items: flex-start;
    box-shadow: none;
  }

  display: flex;
  flex-direction: column;
  
  .main-link {
    display: flex;
    align-items: center;
    transition: all .3s;
    text-decoration: none;
    margin: 0;
    padding: .5rem 1rem .5rem;
    font-size: 1.5rem;
    width: 100%;

    > span {
      transition: transform .3s;
      transform: translateX(-3rem);
    }

    transition: all .3s;

    &.active {
      > span {
        transform: translateX(0rem);
      }

      > ${IconWrapper} {
        transform: translateX(-1rem);
      }
    }
  }
`

const MobileMenuToggle = styled.label`
  padding: 1rem;
  font-size: 2rem;
  
  @media (min-width: ${sizes.m}) {
    display: none;
  }
`

const LinkWrapper = styled.nav`
  background: ${colors.white};
  display: flex;
  flex-direction: column;
  position: fixed;
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  transition: all .3s;
  opacity: 0;
  z-index: -1;

  @media (min-width: ${sizes.m}) {
    width: 100%;
    height: initial;
    position: initial;
    opacity: initial;
    z-index: initial;
  }

  ${MobileMenuToggle} {
    text-align: right;
  }
`

const HiddenCheckbox = styled.input`
  display: none;

  &:checked + ${LinkWrapper} {
    opacity: 1;
    z-index: 99;
  }
`

export default Navigation
