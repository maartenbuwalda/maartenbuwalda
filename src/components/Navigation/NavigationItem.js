import React from 'react'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import Icon from '@material-ui/core/Icon'
import SubNavigation from './SubNavigation'
import { sizes } from '../../constants/mediaQueries'
import colors from '../../constants/colors'

const NavigationItem = ({ theme, item, toggleMenu }) => {
  const currentPath = window.location.hash.replace('#', '')
  return (
    <StyledNavigationItem theme={theme}>
      <NavLink
        exact={item.pathname === '/'}
        to={item.pathname}
        className="main-link"
        activeClassName="active"
        data-test-id={`main-menu-item-${item.pathname}`}
        onClick={() => {
          if (!item.subNavigation.length) toggleMenu(false)
        }}
      >
        <IconWrapper theme={theme}>
          <Icon>{item.icon}</Icon>
        </IconWrapper>
        <span>{item.label}</span>
      </NavLink>
      {item.subNavigation.length > 0 && (
        <SubNavigation
          isToggled={currentPath.includes(item.pathname)}
          toggleMenu={toggleMenu}
          links={item.subNavigation}
        />
      )}
    </StyledNavigationItem>
  )
}

NavigationItem.propTypes = {
  theme: PropTypes.object,
  item: PropTypes.object,
  toggleMenu: PropTypes.func,
}

const IconWrapper = styled.div`
  transition: transform .3s, background-color .6s;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 1rem;
  width: 1rem;
  padding: 1rem;
  color: ${colors.fadedWhite};
  transform: translateX(-4rem) scaleX(0);
  
  @media (min-width: ${sizes.m}) {
    width: 1rem;
    height: 1rem;
    padding: 1rem;
  }
`

const StyledNavigationItem = styled.div`
  .main-link {
    font-family: 'Crete Round', serif;
    text-decoration: none;
    font-size: 1rem;
    color: ${colors.darkGray};
    display: flex;
    align-items: center;
    transition: all .3s;
    margin: 0;
    width: 100%;
    transition: all .3s;

    > span {
      transition: transform .3s;
      transform: translateX(-2rem);
    }


    &.active {
      color: rgba(${({ theme }) => theme.rgb.colorMutedRegular}, 0.8);
      background-color: rgba(${({ theme }) => theme.rgb.colorMutedLight}, 0.2);

      > span {
        transform: translateX(1rem);
      }

      > ${IconWrapper} {
        background-color: ${({ theme }) => theme.hex.colorRegular};
        transform: translateX(0rem);
      }
    }
  }
`

export default NavigationItem
