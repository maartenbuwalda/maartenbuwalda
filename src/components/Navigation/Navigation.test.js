import React from 'react'
import Navigation from './index'
import { componentRenders } from '../../testUtils/rendering'

jest.mock('../../constants/images', () => ({}))

describe('Navigation component', () => {
  const component = <Navigation />
  componentRenders(component)
})
