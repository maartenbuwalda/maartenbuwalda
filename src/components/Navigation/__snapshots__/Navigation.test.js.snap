// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<Card
  content=""
>
  <styled.label
    htmlFor="menu-toggle"
  >
    <WithStyles(Icon)>
      menu
    </WithStyles(Icon)>
  </styled.label>
  <styled.div>
    <styled.input
      id="menu-toggle"
      type="checkbox"
    />
    <styled.nav>
      <styled.label
        htmlFor="menu-toggle"
      >
        <WithStyles(Icon)>
          close
        </WithStyles(Icon)>
      </styled.label>
      <div
        key="0"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="main-link"
          exact={true}
          onClick={[Function]}
          to="/"
        >
          <styled.div>
            <WithStyles(Icon)>
              face
            </WithStyles(Icon)>
          </styled.div>
          <span>
            About me
          </span>
        </NavLink>
      </div>
      <div
        key="1"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="main-link"
          exact={false}
          onClick={[Function]}
          to="/education"
        >
          <styled.div>
            <WithStyles(Icon)>
              book
            </WithStyles(Icon)>
          </styled.div>
          <span>
            Education
          </span>
        </NavLink>
      </div>
      <div
        key="2"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="main-link"
          exact={false}
          onClick={[Function]}
          to="/work"
        >
          <styled.div>
            <WithStyles(Icon)>
              code
            </WithStyles(Icon)>
          </styled.div>
          <span>
            Work
          </span>
        </NavLink>
        <SubNavigation
          closeMenu={[Function]}
          isToggled={false}
          links={
            Array [
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2019 - now: Fontem Ventures",
                "pathname": "/work/fontem-ventures",
                "title": "Fontem Ventures",
              },
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2019 - now: Battlefield Bus Tours",
                "pathname": "/work/battlefield-bus-tours",
                "title": "Battlefield Bus Tours",
              },
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2018 - 2019: Transavia",
                "pathname": "/work/transavia",
                "title": "Transavia",
              },
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2015 - 2017: Wakoopa",
                "pathname": "/work/wakoopa",
                "title": "Wakoopa",
              },
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2017: Stichting Herstelling",
                "pathname": "/work/stichting-herstelling",
                "title": "Stichting Herstelling",
              },
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2017: Go Fort",
                "pathname": "/work/go-fort",
                "title": "Go Fort",
              },
              Object {
                "content": <React.Fragment>
                  <h2>
                    Transavia
                  </h2>
                  <p>
                    Februari 2018 - maart 2019: Transavia
                  </p>
                  <p>
                    Bij Transavia begon ik aan mijn eerste project als gedetacheerde. Ik kwam te werken aan een nieuwe tool die bedoeld is om het proces van groepsboekingen te automatiseren: de Group Booking Solution (GOS). Daarnaast heb ik een grote rol gespeeld in het starten van twee kleinere projecten: het Partner Platform en een React Component Library. Tot slot heb ik bijgedragen aan het oplossen van bugs op transavia.com.
                  </p>
                  <h3>
                    Group Booking Solution (GOS)
                  </h3>
                  <p>
                    De ontwikkeling van GOS was begonnen in november 2017 door beginnende React developers. Ze bouwden een proof of concept om te laten zien dat React de toekomst was voor de front-end van Transavia, en dat is ze gelukt. In het begin werd de keuze gemaakt om het project puur React te houden zodat de developers React konden leren, waardoor er bijvoorbeeld ook geen state management tool geïmplementeerd was. De fundering was daardoor technical debt en daar werd drie maanden op verder gebouwd. Toen ik in februari aan de slag kon, ben ik features blijven toevoegen en alles wat ik daarbij tegenkwam ben ik gaan refactoren om zo de code quality te verbeteren. Uiteindelijk kreeg ik de tijd om het roer om te gooien en Redux aan de applicatie toe te voegen. Ook ben ik gaandeweg unit tests gaan schrijven met Jest en Enzyme, waardoor de code coverage van 0% naar 70% is gestegen.
                  </p>
                  <h3>
                    Het Partner Platform en de React Component Library
                  </h3>
                  <p>
                    De partners van Transavia zijn bijvoorbeeld reisbureaus die een groot aantal boekingen maken voor hun eigen reisaanbod. Voor deze partners zijn verschillende applicaties gebouwd, waaronder GOS. Omdat dit allemaal losse applicaties zijn, ontstond het idee om ze samen te voegen onder het kopje Partner Platform. Dit zou een dashboard worden, van waaruit alle partner applicaties te gebruiken zijn. Het probleem is dat niet alle applicaties met dezelfde technieken zijn gebouwd, en dus niet zo eenvoudig samen te voegen zijn. Een tijdelijke oplossing was om een identieke “navigatie banner” boven elke app te injecteren en op deze manier de apps naar elkaar te laten linken en daarmee het gevoel te geven dat het bestond uit een geheel. Om dit in vier verschillende apps op een effectieve manier te doen, ontstond de behoefte aan een front-end component library. Deze library heb ik vervolgens samen met een andere front-end developer opgezet met webpack, storybook, lerna, react en Azure artifacts. Dit gaf ons uiteindelijk de mogelijkheid om heel simpel het component te importeren, de benodigde data eraan mee te geven en te renderen in elke applicatie.
                  </p>
                </React.Fragment>,
                "image": undefined,
                "label": "2014 - 2015: Dinst",
                "pathname": "/work/dinst",
                "title": "Dinst",
              },
            ]
          }
        />
      </div>
      <div
        key="3"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="main-link"
          exact={false}
          onClick={[Function]}
          to="/contact"
        >
          <styled.div>
            <WithStyles(Icon)>
              chat
            </WithStyles(Icon)>
          </styled.div>
          <span>
            Contact
          </span>
        </NavLink>
      </div>
    </styled.nav>
  </styled.div>
</Card>
`;
